### PR TITLE
GovCloud Support

### PR DESCRIPTION
**Pull request by ryanjpayne**
Changes include:

- AWS GovCloud Support
- AWS Commercial to Falcon GovCloud Support
- Optional Permissions Boundary on IAM Roles
- Update documentation

### CODE REVIEW

File: templates/aws_cspm_cloudformation_eb_comm_gov.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+              "readOnly": [
+                false
+              ]
+            }
+          },
+          "State": "ENABLED",
+          "Targets": [
+            {
+              "Arn": {
+                "Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${AWS::AccountId}:event-bus/default"
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_comm_gov.json:45]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_comm_gov.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              },
+              "RoleArn": {
+                "Fn::Sub": "arn:aws:iam::${AWS::AccountId}:role/CrowdStrikeCSPMEventBridge"
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_comm_gov.json:48]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_comm_gov.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                  }
+                }
+              }
+            ]
+          },
+          "State": "ENABLED",
+          "Targets": [
+            {
+              "Arn": {
+                "Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${AWS::AccountId}:event-bus/default"
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_comm_gov.json:110]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_comm_gov.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+              },
+              "RoleArn": {
+                "Fn::Sub": "arn:aws:iam::${AWS::AccountId}:role/CrowdStrikeCSPMEventBridge"
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_comm_gov.json:113]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_v2.json [hunk 5]
```diff
@@ -52,7 +66,11 @@
           "Targets": [
             {
               "Arn": {
-                "Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${CSAccountNumber}:event-bus/${CSEventBusName}"
+                "Fn::If" : [
+                  "IsGovCloud",
+                  {"Fn::Sub": "arn:aws:events:${AWS::Region}:${CSAccountNumber}:event-bus/${CSEventBusName}"},
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_v2.json:71]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_v2.json [hunk 6]
```diff
@@ -52,7 +66,11 @@
...
+                  {"Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${CSAccountNumber}:event-bus/${CSEventBusName}"}
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_v2.json:72]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_v2.json [hunk 7]
```diff
@@ -117,7 +135,11 @@
           "Targets": [
             {
               "Arn": {
-                "Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${CSAccountNumber}:event-bus/${CSEventBusName}"
+                "Fn::If" : [
+                  "IsGovCloud",
+                  {"Fn::Sub": "arn:aws:events:${AWS::Region}:${CSAccountNumber}:event-bus/${CSEventBusName}"},
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_v2.json:140]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_eb_v2.json [hunk 8]
```diff
@@ -117,7 +135,11 @@
...
+                  {"Fn::Sub": "arn:aws:events:${DefaultEventBusRegion}:${CSAccountNumber}:event-bus/${CSEventBusName}"}
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_eb_v2.json:141]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_v2.json [hunk 9]
```diff
@@ -140,6 +150,13 @@
         }
       },
       "Properties": {
+        "PermissionsBoundary": { 
+            "Fn::If": [ 
+              "SetPermissionsBoundary", 
+              {"Fn::Sub": "arn:aws:iam::${AWS::AccountId}:policy/${PermissionsBoundary}"},
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_v2.json:156]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_v2.json [hunk 10]
```diff
@@ -224,6 +245,13 @@
...
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_v2.json:251]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_v2.json [hunk 11]
```diff
@@ -299,6 +329,13 @@
...
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_v2.json:335]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

File: templates/aws_cspm_cloudformation_v2.json [hunk 12]
```diff
@@ -370,6 +407,13 @@
...
```
Comment (kkvinjam) [inline templates/aws_cspm_cloudformation_v2.json:413]: Replace `:aws:` with `${AWS::Partition}`. This is required to support govcloud.

**Review by kkvinjam — CHANGES_REQUESTED** [review]

File: templates/aws_cspm_cloudformation_eb_comm_gov.json (hunk 4 above)
Comment (ryanjpayne) [reply templates/aws_cspm_cloudformation_eb_comm_gov.json:113]: this template only used for commercial

File: templates/aws_cspm_cloudformation_eb_comm_gov.json (hunk 3 above)
Comment (ryanjpayne) [reply templates/aws_cspm_cloudformation_eb_comm_gov.json:110]: this template only used for commercial

File: templates/aws_cspm_cloudformation_eb_comm_gov.json (hunk 2 above)
Comment (ryanjpayne) [reply templates/aws_cspm_cloudformation_eb_comm_gov.json:48]: this template only used for commercial

File: templates/aws_cspm_cloudformation_eb_comm_gov.json (hunk 1 above)
Comment (ryanjpayne) [reply templates/aws_cspm_cloudformation_eb_comm_gov.json:45]: this template only used for commercial

**Comment by kkvinjam** [comment]
> E2E test has completed with errors. If you are an external contributor, please contact the project maintainers for more information.

Test bed cleanup issue.

**Comment by kkvinjam** [comment]
> E2E test has completed with errors. If you are an external contributor, please contact the project maintainers for more information.

Still fails with previously reported issue. @ryanjpayne 

`        "errors": [
            {
                "code": 401,
                "message": "access denied, invalid bearer token"
            }
        ]`

**Comment by kkvinjam** [comment]
> E2E test has completed with errors. If you are an external contributor, please contact the project maintainers for more information.

Failure related to the new code under certain conditions. Shared the details with author.

**Review by awsdmitry — APPROVED** [review]
Ran through the stacks in both commercial (to govcloud endpoint) and govcloud. Approved with the caveat that the commercial to govcloud stack must be run in us-east-1 now. Should be noted somewhere in the release notes.

**Comment by ryanjpayne** [comment]
> Ran through the stacks in both commercial (to govcloud endpoint) and govcloud. Approved with the caveat that the commercial to govcloud stack must be run in us-east-1 now. Should be noted somewhere in the release notes.

addressed this in latest commit

**Review by kkvinjam — APPROVED** [review]

**Comment by kkvinjam** [comment]
> E2E test has completed with errors. If you are an external contributor, please contact the project maintainers for more information.

This is test platform related issue. Will resume the tests once its fixed.

**Pull request merged**